src/misc.jl

```diff
@@ -30,31 +29,53 @@ function laplacian(con_mat::AbstractMatrix{<:Real})
     return Symmetric(lap, :L)
 end
 
+function laplacian(con_mat::AbstractMatrix{Int})
+    return laplacian(con_mat .> 0) # treating the substation graph as a simple graph.
+end
+
+# Shorter code but worse performance.
+# function laplacian(con_mat::AbstractMatrix{<:Bool})
+#     return sum(con_mat, 1) .* eye(Int, size(con_mat)[1]) - con_mat
+# end
+
 """
-    total_links(con_mat::AbstractMatrix{<:Real})
+    total_links(con_mat::AbstractMatrix{Bool})
 
 Return the total number of links in a system with connectivity matrix `con_mat`.
 """
-function total_links(con_mat::AbstractMatrix{<:Real})
-    return (1 / 2) * trace(laplacian(con_mat))
+function total_links(con_mat::AbstractMatrix{Bool})
+    return (1 / 2) * sum(con_mat)
+end
+
+"""
+    total_links(con_mat::AbstractMatrix{Integer})
+
+Return the total number of links in a system with connectivity matrix `con_mat`. The graph
+will be treated as a simple graph.
+"""
+function total_links(con_mat::AbstractMatrix{Int})
```

Comment:
Same issue. You can still use `<:Real`, but just use `zero(T)` in the comparison.